### PR TITLE
Add the default number indicator at the View Details too

### PR DESCRIPTION
**Pull request by pavelpoley**
- Added the default number indicator at the view details
- For duplicated contacts, try to reuse default number
- Update the default number in all contact sources

### CODE REVIEW

File: app/src/main/kotlin/com/simplemobiletools/contacts/pro/activities/ViewContactActivity.kt [hunk 1]
```diff
@@ -45,6 +46,10 @@ class ViewContactActivity : ContactActivity() {
     private var contactSources = ArrayList<ContactSource>()
     private var showFields = 0
     private var fullContact: Contact? = null    // contact with all fields filled from duplicates
+    private var duplicateInitialized = false
+    private val mergeDuplicate: Boolean by lazy {
```
Comment (tibbi) [inline app/src/main/kotlin/com/simplemobiletools/contacts/pro/activities/ViewContactActivity.kt:50]: dont use lazy please, it is too confusing

**Comment by tibbi** [comment]
when a default number is starred on View screen, then I open the Edit one and go Back at the view one, the star blinks (disappears, then reappears). The contact sources at the bottom dont blink though, so lets try making the star not blink either.

**Comment by tibbi** [comment]
another note, would be good to load the star in synchronously right after loading in the contact so the star doesnt appear delayed. I know that it can potentially conflict with something if other contact sources load in, but that should be really rare. The vast majority of people should have 1 number set as the default one, no matter how many contact sources is the contact available in.

**Comment by pavelpoley** [comment]
loading the star synchronously (fetch duplicates to reuse default number), seems too much work on the main thread and will appear in delay anyway, and some part of the chain must run on background thread.

pushed a few changes, the star will appear fine in standard cases.
may blink when there is a conflict or when "merge duplicate" setting is disabled and we select contact without default number, but his duplicate contact has default number.

**Comment by tibbi** [comment]
we do not fetch duplicates synchronously, just show the default number at the initial loaded contact. Will test it later.

**Comment by tibbi** [comment]
yeah, this looks good to me. It should work just fine in 99% of cases, no need to take care of some advanced edge cases.
